kernel/src/hil/adc.rs

```diff
@@ -1,14 +1,20 @@
-//! Simplest attempt at an ADC interface.
-//!
-//!  Author: Philip Levis <pal@cs.stanford.edu>
-//!  Date: June 10, 2015
-//!
-
-pub trait Request {
-    fn sample_done(&self, val: u16, request: &'static Request);
+pub trait Client {
+    fn sample_done(&self, sample: u16);
 }
 
-pub trait AdcInternal {
-    fn initialize(&'static mut self) -> bool;
-    fn sample(&self, channel: u8, callback: &'static Request) -> bool;
+pub trait AdcSingle {
+    fn initialize(&self) -> bool;
+    fn sample(&self, channel: u8) -> bool;
+    // fn cancel_sample(&self) -> bool;
 }
+
+// pub trait Frequency {
```

Comment:
So I started with what Phil proposed and went from there. What is commented out is the proposed HIL for a continuous reading trait, which might be nice if we ever want to sample faster than 3kHz. I think we can get rid of the commented code if the continuous mode isn't a priority.
Additionally, the initialize function needs to be expanded and actually provide configuration options.